### PR TITLE
Moving from supertest to axios

### PR DESCRIPTION
**Pull request by goldbergyoni**
@mikicho @DanielGluskin Once you approve this (draft PR), we can apply to all files. Right not, any other file than basic-testing will fail because of this breaking change

### CODE REVIEW

File: example-application/test/basic-tests.test.js [hunk 1]
```diff
@@ -39,33 +49,34 @@ afterAll(async (done) => {
 // ️️️✅ Best Practice: Structure tests
 describe('/api', () => {
   describe('GET /order', () => {
-    test('When asked for an existing order, Then should retrieve it and receive 200 response', async () => {
+    test('When asked for an existing order, Then should retrieve it and receive 200 response loo', async () => {
```
Comment (mikicho) [inline example-application/test/basic-tests.test.js:52]: typo?

File: example-application/test/basic-tests.test.js [hunk 2]
```diff
@@ -39,33 +49,34 @@ afterAll(async (done) => {
...
       const orderToAdd = {
         userId: 1,
         productId: 2,
         mode: 'approved',
       };
       const {
-        body: { id: addedOrderId },
-      } = await request(expressApp).post('/order').send(orderToAdd);
+        data: { id: addedOrderId },
+      } = await axiosAPIClient.post(`/order`, orderToAdd);
```
Comment (mikicho) [inline example-application/test/basic-tests.test.js:61]:
```suggestion
      } = await axiosAPIClient.post('/order', orderToAdd);
```

**Review by mikicho — APPROVED** [review]

**Comment by mikicho** [comment]
@goldbergyoni some anti-patterns tests don't pass. not sure if it's intended. maybe we need to add `@expectToFail` tag to every test that we expect it to fail? maybe we should test it fail? :)

**Review by goldbergyoni — COMMENTED** [review]
> @goldbergyoni some anti-patterns tests don't pass. not sure if it's intended. maybe we need to add `@expectToFail` tag to every test that we expect it to fail? maybe we should test it fail? :)

@mikicho All tests pass on my machine, Jest config has '!**/*anti-pattern*'

I like your ideas but not familiar with a way to make it happen using Jest, do I miss some feature?

File: example-application/test/basic-tests.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (mikicho) [reply example-application/test/basic-tests.test.js]: can't find this piece of code.

**Comment by mikicho** [comment]
> All tests pass on my machine, Jest config has '!**/_anti-pattern_'

Yes, but what happens if we run them? they still should pass, I think.

> I like your ideas but not familiar with a way to make it happen using Jest, do I miss some feature?

Just a simple string, to let us know we expect this test should fail and everything is ok

**Pull request merged**